library/std/src/thread/mod.rs

```diff
@@ -1445,8 +1445,15 @@ impl<T> JoinHandle<T> {
 
     /// Checks if the associated thread is still running its main function.
     ///
+    /// `is_running` supports implementing a non-blocking join operation, by
+    /// checking `is_running`, and calling `join` if it returns `false`.
+    /// `is_running` is not intended to be tightly polled in a busyloop or
+    /// similar; use `join` to block waiting on another thread.
```

Comment:
Without explaining what exactly counts as a 'busyloop', this might be a bit confusing. It's entirely fine to have something like `while t.is_running() { very_small_amount_of_work() }` if you want to run some calculations but stop when another thread is done. (`is_running()` is very efficient, as it just checks an atomic. Similar to any other atomic stop flag.)

Maybe it's better to just say that `join` should be used to wait/block?

```suggestion
    /// `is_running` does not block.
    /// To block while waiting on another thread to finish, use [`join`][Self::join] instead.
```